vent/sqldb/utils.go

```diff
@@ -388,39 +388,36 @@ func (db *SQLDB) getBlockTables(eventFilter, block string) (types.EventTables, e
 }
 
 // getUpsertParams builds parameters in preparation for an upsert query
-func getUpsertParams(upsertQuery types.UpsertQuery, row types.EventDataRow) ([]interface{}, string, error) {
+func getUpsertParams(upsertQuery types.UpsertQuery, row types.EventDataRow) ([]interface{}, error) {
 	pointers := make([]interface{}, upsertQuery.Length)
-	containers := make([]sql.NullString, upsertQuery.Length)
+
 
 	for colName, col := range upsertQuery.Columns {
 		// interface=data
-		pointers[col.InsPosition] = &containers[col.InsPosition]
-		if col.UpdPosition > 0 {
-			pointers[col.UpdPosition] = &containers[col.UpdPosition]
-		}
 
 		// build parameter list
 		if value, ok := row[colName]; ok {
 			// column found (not null)
-			containers[col.InsPosition] = sql.NullString{String: value, Valid: true}
+			pointers[col.InsPosition] = &value
 
 			// if column is not PK
 			if col.UpdPosition > 0 {
-				containers[col.UpdPosition] = sql.NullString{String: value, Valid: true}
+				pointers[col.UpdPosition] = &value
 			}
 		} else if col.UpdPosition > 0 {
 			// column not found and is not PK (null)
-			containers[col.InsPosition].Valid = false
-			containers[col.UpdPosition].Valid = false
+			null:=sql.NullString{ Valid: false}
+			pointers[col.InsPosition] = &null
```

Comment:
Does it work if you change this to just `pointers[col.InsPosition] = nil` ?

Reply:
function no longer exist